syntax = "proto3";
package profastbit;
service dump_fastbit {
  rpc dumptofast(prodata) returns (datareply) {}
}
message datareply{
	int32 ret = 1;
}
message prodata{
	string dump_dir = 1;
	int32 num_entries = 2;
	int32 fb_elementlen = 3;
	repeated int32 fastbit_dump_switch = 4;
	repeated TemplateElementId templateelement = 5;
	repeated fb_element fb_element = 6;
}
message TemplateElementId{
	int32 templateElementLen = 1;
	string netflowElementName = 2;
}
message fb_element{
	repeated bytes fb_element_data = 1;
}
